src/main/java/org/killbill/billing/plugin/adyen/client/AdyenConfigProperties.java

```diff
@@ -90,22 +94,29 @@
 
     private final Period pending3DsPaymentExpirationPeriod;
 
-    public AdyenConfigProperties(final Properties properties) {
+    private final String currentRegion;
+
+    public AdyenConfigProperties(final Properties properties, final Properties globalProperties) {
+        this.currentRegion = PluginEnvironmentConfig.getRegion(globalProperties);
```

Comment:
> The main benefit being we wouldn't need to work with the cumbersome map like syntax.

Yeah, I just went with the same pattern (let's make it _consistently_ painful!), hoping that a YAML version will come soon. But maybe it's better to start improving what we have already.

See 8df0d2e4c59e0af005033462962c263d7b570ec7 and b020102fd63d015f50a89e28ac08b46d987ca67c.